File: backend/app/models/step_log.py
"""
StepLog document.

One document per user per day — upserted so today's entry is always current.
"""
from __future__ import annotations

from datetime import date
from uuid import UUID

from pymongo import ASCENDING

from app.models.base import BaseDocument


class StepLog(BaseDocument):
    user_id: UUID
    date: date
    steps: int

    class Settings:
        name = "step_logs"
        indexes = [
            # Unique per user per date
            [("user_id", ASCENDING), ("date", ASCENDING)],
        ]

    def __repr__(self) -> str:
        return f"<StepLog user={self.user_id} date={self.date} steps={self.steps}>"

File: mobile/lib/core/models/step_log.dart
class StepLog {
  final String id;
  final DateTime date;
  final int steps;

  const StepLog({required this.id, required this.date, required this.steps});

  factory StepLog.fromJson(Map<String, dynamic> json) => StepLog(
        id: json['id'] as String,
        date: DateTime.parse(json['date'] as String),
        steps: json['steps'] as int,
      );
}
